qiskit/algorithms/amplitude_estimators/ae.py

```diff
@@ -348,6 +355,13 @@ def estimate(self, estimation_problem: EstimationProblem) -> "AmplitudeEstimatio
         if estimation_problem.objective_qubits is None:
             raise ValueError("The objective_qubits property of the estimation problem must be set.")
 
+        if estimation_problem._is_good_state is not None:
```

Comment:
As others could build other AE algorithms using the problem, rather than using a private var here to detect this, should we rather add some public method on the problem say. has_good_state to return if the user provided a good state callback or if is_good_state is just defaulting it? Maybe this here is an exception and using the private var with the same sub-package is ok.

Reply:
Yeah it might be nice to allow querying that property. I'm not sure if it would be a bit overkill here, since we're not aware of any other use-cases so far, but we could still add it.

Reply:
It would avoid using a private member of the problem - I grant you some public method is unlikely to be used elsewhere, and the likelihood of the private implementation in the problem changing is minimal. Hence I concluded above with "Maybe this here is an exception and using the private var with the same sub-package is ok." I guess I'm still not so keen on it, but it works and if this is your preference...

Reply:
Added in ee935b5 🙂